### PR TITLE
Initial sketch of a FastAPI server

### PR DESCRIPTION
**Pull request by alesgenova**

### CODE REVIEW

File: server/server/main.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+    try:
+        while True:
+            _ = await socket.receive_json()
+    except WebSocketDisconnect:
+        websocket_ids.remove(websocket.id)
+
+@app.get("/")
+async def get_index():
+    index_file = f"{os.path.join(os.path.dirname(__file__), 'index.html')}"
```
Comment (psavery) [inline server/server/main.py:194]: I think this is a little easier to do with `pathlib`'s `with_name()` function. For instance:

```python
>>> from pathlib import Path
>>> Path('/path/to/file.py').with_name('index.html')
PosixPath('/path/to/index.html')
```

Same for [here](https://github.com/OpenChemistry/oremda/pull/20/files#diff-b7e71dbe9962684ab6f05be10a8404725062a2a18836cba09decea66fc2aa169R200).

File: oremda/pipeline.py [hunk 2]
```diff
@@ -154,6 +160,8 @@ def run(self):
         )
         run_operators = set()
 
+        self.observer.on_start(self)
```
Comment (psavery) [inline oremda/pipeline.py:163]: Is this observer always associated with this pipeline? If that's the case, maybe you should pass in `self` in the `PipelineObserver.__init__()` method and store it inside, so you don't have to pass in `self` for all of the observer functions.

**Review by psavery — APPROVED** [review]
LGTM

File: oremda/pipeline.py (hunk 2 above)
Comment (alesgenova) [reply oremda/pipeline.py:163]: I'm not quite set on the API of the PipelineObserver, but the reasoning is that we could use a single instance of the observer to observe multiple pipelines (for example, all the pipelines associated with a single client/session)

**Pull request merged**